src/toxcore/messenger/packet/status_message.rs

```diff
@@ -0,0 +1,88 @@
+/*! StatusMessage struct.
+It is used to send my status message to a friend.
+This packet is sent to my friends every time they become online, or whenever my status message is changed.
+*/
+
+use std::str;
+use nom::rest;
+
+use crate::toxcore::binary_io::*;
+
+/// Maximum size in bytes of status message string
+const MAX_STATUS_MESSAGE_DATA_SIZE: usize = 1007;
+
+/** StatusMessage is a struct that holds string of my status message.
+
+This packet is used to transmit sender's status message to a friend.
+Every time a friend become online or my status message is changed,
+this packet is sent to the friend or to all friends of mine.
+
+Serialized form:
+
+Length    | Content
+--------- | ------
+`1`       | `0x31`
+`0..1007`  | UTF8 byte string
+
+*/
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct StatusMessage(String);
```

Comment:
How about unifying with https://github.com/tox-rs/tox/pull/318/files#diff-8b40c8278e389268ea076fb2aaeb6d84R39 - you used short form of struct here, but full there.